internal/core/src/segcore/SegmentGrowingImpl.h

```diff
@@ -225,7 +225,13 @@ class SegmentGrowingImpl : public SegmentGrowing {
 
     bool
     HasIndex(FieldId field_id) const override {
-        return true;
+        auto& field_meta = schema_->operator[](field_id);
+        if (datatype_is_vector(field_meta.get_data_type()) &&
+            segcore_config_.get_enable_growing_segment_index()) {
+            return true;
+        }
+
+        return false;
```

Comment:
Growing segment index built or not can not just be statically determined by the config. For growing scene, index could be built when data has been accumulated to a certain scale. Moreover, user creating FLAT index will disable growing index for 100% recall. You could use `SyncDataWithIndex` to determine it, which means all growing data have been inserted into index, then all search/getVectors operation will be dispatched to the index.